rust/src/record_iterator.rs

```diff
@@ -0,0 +1,83 @@
+//! Ergonomic wrapper for reading and parsing records from a [`std::io::Read`] implementation.
+use crate::{
+    lexer::{LexError, Lexer},
+    parse::parse_record,
+    records::{Chunk, Record},
+};
+use lifetime::IntoStatic;
+use std::io::Read;
+
+/// A wrapper for a [`std::io::Read`] implementation which parses out
+/// records and returns them from an iterator.
+pub struct RecordIterator<'a, R: std::io::Read> {
+    lexer: Lexer<'a, R>,
+    chunk_lexer: Option<Lexer<'a, std::io::Cursor<Vec<u8>>>>,
+    raw_record: Vec<u8>,
+    expand_chunks: bool,
```

Comment:
are these public by default? I notice the struct is marked pub, bu the `impl Iterator` and `fn next` are not, which was a little surprising

Reply:
you don't have to mark functions as pub when you're implementing a trait. Struct fields are private by default.